src/main/java/frc/trigon/robot/commands/factories/AutonomousCommands.java

```diff
@@ -0,0 +1,91 @@
+import org.trigon.utilities.mirrorable.MirrorableRotation2d;
+
+import java.util.Optional;
+import java.util.function.Supplier;
+
+/**
+ * A class that contains commands that are used during the 15-second autonomous period at the start of each match.
+ */
+public class AutonomousCommands {
+    private static final ShootingCalculations SHOOTING_CALCULATIONS = ShootingCalculations.getInstance();
+    private static final ObjectDetectionCamera NOTE_DETECTION_CAMERA = CameraConstants.NOTE_DETECTION_CAMERA;
+
+    public static Command getResetPoseToAutoPoseCommand(Supplier<String> pathName) {
+        return new InstantCommand(
+                () -> {
+                    if (DriverStation.isEnabled())
+                        return;
+                    final Pose2d autoStartPose = PathPlannerAuto.getStaringPoseFromAutoFile(pathName.get());
+                    RobotContainer.POSE_ESTIMATOR.resetPose(new MirrorablePose2d(autoStartPose, true).get());
+                }
+        ).ignoringDisable(true);
+    }
+
+    public static Command getNoteCollectionCommand() {
+        return new ParallelCommandGroup(
+                IntakeCommands.getSetTargetStateCommand(IntakeConstants.IntakeState.COLLECT).unless(RobotContainer.INTAKE::hasNote),
+                IntakeCommands.getSetTargetStateCommand(IntakeConstants.IntakeState.STOP).onlyIf(RobotContainer.INTAKE::hasNote)
+        );
+    }
+
+    public static Command getPrepareForShooterEjectionCommand() {
+        return new ParallelCommandGroup(
+                PitcherCommands.getSetTargetPitchCommand(ShootingConstants.EJECT_FROM_SHOOTER_PITCH),
+                ShooterCommands.getSetTargetVelocityCommand(ShootingConstants.EJECT_FROM_SHOOTER_VELOCITY_ROTATIONS_PER_SECOND)
+        );
+    }
+
+    public static Command getFeedNoteCommand() {
+        return new ParallelCommandGroup(
+                IntakeCommands.getSetTargetStateCommand(IntakeConstants.IntakeState.FEED_SHOOTING),
+                GeneralCommands.getVisualizeNoteShootingCommand()
+        );
+    }
+
+    public static Command getAlignToSpeakerCommand() {
+        return new ExecuteEndCommand(
+                () -> overrideRotation(Optional.of(SHOOTING_CALCULATIONS.getTargetShootingState().targetRobotAngle().get())),
+                () -> {
+                    if (!RobotContainer.INTAKE.hasNote())
+                        overrideRotation(Optional.empty());
+                }
+        );
+    }
+
+    public static Command getAlignToNoteCommand() {
+        return new InitExecuteCommand(
+                NOTE_DETECTION_CAMERA::startTrackingObject,
+                () -> overrideRotation(Optional.of(Rotation2d.fromDegrees(NOTE_DETECTION_CAMERA.getTrackedObjectYaw())))
+        ).andThen(new InstantCommand(() -> {
+                    if (RobotContainer.SWERVE.atAngle(new MirrorableRotation2d(NOTE_DETECTION_CAMERA.getTrackedObjectYaw(), true)))
+                        overrideRotation(Optional.empty());
```

Comment:
1. This command will never reach the andThen
2. Why the atAngle?
3. Use another thing for a command that does something when it starts, onExecute, and onEnd. It's called a FunctionalCommand.